src/utils/parseHTML.ts

```diff
@@ -38,9 +39,12 @@ const parseHTML = (html: string): ReactElement | ReactElement[] | string => {
                     return rewriteHeader(domNode);
                 }
             }
-            if (domNode.data) {
+            if (domNode.data && domNode.parent?.name !== 'code') {
```

Comment:
This is the main thing - only render the special elements if they're not inside a code block.